lib/devise/strategies/jwt_authenticatable.rb

```diff
@@ -6,13 +6,13 @@ module Devise
   module Strategies
     class JwtAuthenticatable < Authenticatable
       def valid?
-        super || token # Why super? Should always return false
+        token
       end
 
       def authenticate!
         Rails.logger.debug("Authenticating with JWT.")
-        return fail(:invalid_token) if @token.blank? # Is a more sophisticated check required here?
-        decoded_token = token_data(@token)
+        return fail(:invalid_token) if token.blank? # Is a more sophisticated check required here?
```

Comment:
If more sophisticated check if needed there than please add it, remove comment otherwise